### ISSUE TITLE
Hi, I cant see CallKit UI in case if app is in background iOS 12

### ISSUE THREAD

**Issue by narekyan** [twilio/video-quickstart-ios#358]
Hi, everything works, for other cases, beside on this case.
I have standard call application which use twilio.
When I make call to the device I can see CallKit UI when app is

iOS 11
1) in foreground
2) in background 
3) in killed state

iOS 12
1) in foreground
2) in background I DONT see CallKit UI
3) in killed state

So problem is only for background state on iOS 12. So user cant accept or decline call.

Is there anyone who had fixed this issue?
Can you help me?

**Comment by narekyan** [comment]
pls help

**Comment by paynerc** [comment]
@narekyan,

If I am understanding this correctly, you are getting the incoming notification, passing it off to the `CXProvider` `reportNewIncomingCall` method and it's not displaying the CallKit UI on iOS 12, yet it works on iOS 11?

What devices are you running this on that is exhibiting this behavior? What Background Modes do you have set in your application?

Have you tried running the `VideoCallKitQuickStart` quickstart application and simulating an incoming call? I have run this on iOS 12 while the app is in the background and with the phone being locked and the CallKit incoming call UI appears.

Ryan

**Comment by narekyan** [comment]
@paynerc  
Yes , we use ipad ios 11 and iphone6s ios 12
All cases work on both device, beside
the only case that does not work is when receiving incoming call when app is in background on iphone 6s ios 12
we have 3 background modes - Audio, VoIP, Remote nots

**Comment by narekyan** [comment]
@paynerc 
self.callKitProvider.reportNewIncomingCall(with: uuid, update: callUpdate) { error in
           // SO this callback is not called when app is in background, but works on when in foreground (only ios 12 problem)
        }

**Comment by paynerc** [comment]
@narekyan,

How are you notifying the application that there is an incoming call?

Ryan

**Comment by paynerc** [comment]
@narekyan,

It's been a bit since we last spoke. Have you found a solution to your issue?

Thank you!

Ryan

**Comment by narekyan** [comment]
@paynerc I dont remember alrady, and now i dont even work on that project))

**Comment by CaptainJeff** [comment]
@paynerc did you ever find a solution to this? I'm facing the same issue

**Comment by paynerc** [comment]
@CaptainJeff,

No, I never heard back on the initial report to understand if this is indeed still an issue. How are you getting notifications to your app about an incoming call?

Ryan

**Comment by CaptainJeff** [comment]
Damn. So I have this issue on older iphones it seems. This works on iphone X and iphone XR just fine (ios 13) but on iPhone SE (13.3.1) and iPhone 7 (12.4.6) I get this issue. I'm using a VOIP notification to listen for the notification and immediately register callkit before calling completion on the notification. 

Occasionally I'll get an error like this which will cause a crash

```
Exception Type:  EXC_CRASH (SIGKILL)
Exception Codes: 0x0000000000000000, 0x0000000000000000
Exception Note:  EXC_CORPSE_NOTIFY
Termination Reason: Namespace SPRINGBOARD, Code 0x8badf00d
Termination Description: SPRINGBOARD, scene-create watchdog transgression: com.x.x exhausted real (wall clock) time allowance of 19.25 seconds |  | ProcessVisibility: Foreground | ProcessState: Running | WatchdogEvent: scene-create | WatchdogVisibility: Foreground | WatchdogCPUStatistics: ( | "Elapsed total CPU time (seconds): 9.680 (user 9.680, system 0.000), 70% CPU", | "Elapsed application CPU time (seconds): 5.934, 43% CPU" | )
Triggered by Thread:  0
```

**Comment by appdoctorinfo** [comment]
Hi @paynerc, how should I do to get incoming call notification? Is there any support from Twilio?
